spring-integration-core/src/main/resources/META-INF/spring.integration.default.properties

```diff
@@ -4,3 +4,4 @@ spring.integration.channels.maxBroadcastSubscribers=0x7fffffff
 spring.integration.taskScheduler.poolSize=10
 spring.integration.messagingTemplate.throwExceptionOnLateReply=false
 spring.integration.messagingAnnotations.require.componentAnnotation=false
+spring.integration.messagingGateway.convertReceiveMessage=false
```

Comment:
Your `if...else` looks wrong.
I tried to figure out the solution without the property, but it isn't going to go just with simple `if...else`.
We are expecting `Message<?>` but don't have control over `gateway.receive()`. Therefore always only `payload` independently of its `Message`. That's already target application responsibility to wrap message to another, e.g. to retain the headers for further logic.
I'm changing the property to `true` with the hope that no one will explore it. And that is much easier to me to fix my app, rather than answer to all those annoying questions like "Why have you fixed that bug?!"